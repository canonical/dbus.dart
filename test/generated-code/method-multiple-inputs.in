<node>
  <interface name="com.example.Test">
    <method name="Hello">
      <arg name="byte_value" type="y" direction="in"/>
      <arg name="boolean_value" type="b" direction="in"/>
      <arg name="int16_value" type="n" direction="in"/>
      <arg name="uint16_value" type="q" direction="in"/>
      <arg name="int32_value" type="i" direction="in"/>
      <arg name="uint32_value" type="u" direction="in"/>
      <arg name="int64_value" type="x" direction="in"/>
      <arg name="uint64_value" type="t" direction="in"/>
      <arg name="double_value" type="d" direction="in"/>
      <arg name="string_value" type="s" direction="in"/>
      <arg name="object_path_value" type="o" direction="in"/>
      <arg name="signature_value" type="g" direction="in"/>
      <arg name="variant_value" type="v" direction="in"/>
      <arg name="unix_fd_value" type="h" direction="in"/>
      <arg name="struct_value" type="(si)" direction="in"/>
      <arg name="array_value_y" type="ay" direction="in"/>
      <arg name="array_value_b" type="ab" direction="in"/>
      <arg name="array_value_n" type="an" direction="in"/>
      <arg name="array_value_q" type="aq" direction="in"/>
      <arg name="array_value_u" type="ai" direction="in"/>
      <arg name="array_value_n" type="au" direction="in"/>
      <arg name="array_value_x" type="ax" direction="in"/>
      <arg name="array_value_t" type="at" direction="in"/>
      <arg name="array_value_d" type="ad" direction="in"/>
      <arg name="array_value_s" type="as" direction="in"/>
      <arg name="array_value_o" type="ao" direction="in"/>
      <arg name="array_value_g" type="ag" direction="in"/>
      <arg name="array_value_v" type="av" direction="in"/>
      <arg name="array_value_h" type="ah" direction="in"/>
      <arg name="array_value_struct" type="a(si)" direction="in"/>
      <arg name="array_value_array" type="aas" direction="in"/>
      <arg name="dict_value_sv" type="a{sv}" direction="in"/>
      <arg name="dict_value_us" type="a{us}" direction="in"/>
      <arg name="dict_value_dict" type="a{sa{sv}}" direction="in"/>
    </method>
  </interface>
</node>
