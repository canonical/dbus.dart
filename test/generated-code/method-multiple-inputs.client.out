import 'package:dbus/dbus.dart';

class ComExampleTest extends DBusRemoteObject {
  ComExampleTest(DBusClient client, String destination, DBusObjectPath path) : super(client, name: destination, path: path);

  /// Invokes com.example.Test.Hello()
  Future<void> callHello(int byte_value, bool boolean_value, int int16_value, int uint16_value, int int32_value, int uint32_value, int int64_value, int uint64_value, double double_value, String string_value, String object_path_value, DBusSignature signature_value, DBusValue variant_value, DBusStruct struct_value, List<int> array_value_y, List<bool> array_value_b, List<int> array_value_n, List<int> array_value_q, List<int> array_value_u, List<int> array_value_n, List<int> array_value_x, List<int> array_value_t, List<double> array_value_d, List<String> array_value_s, List<String> array_value_o, List<DBusSignature> array_value_g, List<DBusValue> array_value_v, List<DBusStruct> array_value_struct, List<List<String>> array_value_array, Map<String, DBusValue> dict_value_sv, Map<int, String> dict_value_us, Map<String, Map<String, DBusValue>> dict_value_dict, {bool noAutoStart = false, bool allowInteractiveAuthorization = false}) async {
    await callMethod('com.example.Test', 'Hello', [DBusByte(byte_value), DBusBoolean(boolean_value), DBusInt16(int16_value), DBusUint16(uint16_value), DBusInt32(int32_value), DBusUint32(uint32_value), DBusInt64(int64_value), DBusUint64(uint64_value), DBusDouble(double_value), DBusString(string_value), DBusObjectPath(object_path_value), signature_value, DBusVariant(variant_value), struct_value, DBusArray.byte(array_value_y), DBusArray(DBusSignature('b'), array_value_b.map((child) => DBusBoolean(child))), DBusArray.int16(array_value_n), DBusArray.uint16(array_value_q), DBusArray.int32(array_value_u), DBusArray.uint32(array_value_n), DBusArray.int64(array_value_x), DBusArray.uint64(array_value_t), DBusArray.double(array_value_d), DBusArray.string(array_value_s), DBusArray.objectPath(array_value_o), DBusArray(DBusSignature('g'), array_value_g.map((child) => child)), DBusArray.variant(array_value_v), DBusArray(DBusSignature('(si)'), array_value_struct.map((child) => child)), DBusArray(DBusSignature('as'), array_value_array.map((child) => DBusArray.string(child))), DBusDict.stringVariant(dict_value_sv), DBusDict(DBusSignature('u'), DBusSignature('s'), dict_value_us.map((key, value) => MapEntry(DBusUint32(key), DBusString(value)))), DBusDict(DBusSignature('s'), DBusSignature('a{sv}'), dict_value_dict.map((key, value) => MapEntry(DBusString(key), DBusDict.stringVariant(value))))], replySignature: DBusSignature(''), noAutoStart: noAutoStart, allowInteractiveAuthorization: allowInteractiveAuthorization);
  }
}
